### PR TITLE
Fix for phantomjs evaluation error on missing variable and undefined …

### PR DESCRIPTION
**Pull request by bsawyer**
Fix for phantomjs evaluation error on missing variable and undefined property of window.

### CODE REVIEW

File: notify.js [hunk 1]
```diff
@@ -102,7 +102,7 @@
     };
 
     // true if the permission is not granted
-    Notify.needsPermission = (Notification.permission === 'granted') ? false : true;
+    Notify.needsPermission = (window.Notification.permission && window.Notification.permission === 'granted') ? false : true;
```
Comment (alexgibson) [inline notify.js]: I think you can shorted `window` here to just `w`?

**Comment by alexgibson** [comment]
Thanks for this - there are some other occurrences in the file where this would need applying too. It would be great if you could update those too?

**Comment by bsawyer** [comment]
Sure thing; If it makes sense I'll set a local var in the parent closure e.g. N = w.Notification to test off of.

**Comment by alexgibson** [comment]
Looks great thanks! I'll squash your two commits and merge this shortly.

**Comment by alexgibson** [comment]
Squashed and merged in ee390d506cac35b3167950d4e81a866b459c195c

Thanks for the PR! :+1:

**Pull request closed without merging**